packages/react-router/modules/Prompt.js

```diff
@@ -39,9 +44,28 @@ class Prompt extends React.Component {
     }
   }
 
+  onbeforeunload(e) {
+    var dialogText = "Changes you made may not be saved."
+    e.returnValue = dialogText
+    return dialogText
+  }
+
+  enableUnload(_onbeforeunload) {
+    if(this._subscribed) { return }
+    this._subscribed = _onbeforeunload ? _onbeforeunload : this.onbeforeunload;
+    window.addEventListener(BEFOREUNLOAD, this._subscribed)
```

Comment:
`<Prompt>` should work everywhere, `<BrowserPrompt>` is a `<Prompt>` that is able to do stuff with the `beforeunload` DOM event. That's how I've thought about it anyway. I'm open to suggestions if you've got different ideas about how it should be done.